presto-main/src/main/java/com/facebook/presto/type/DateOperators.java

```diff
@@ -104,12 +104,17 @@ public static boolean between(@SqlType(StandardTypes.DATE) long value, @SqlType(
     @SqlType(StandardTypes.TIMESTAMP)
     public static long castToTimestamp(ConnectorSession session, @SqlType(StandardTypes.DATE) long value)
     {
-        long utcMillis = TimeUnit.DAYS.toMillis(value);
+        if (session.isLegacyTimestamp()) {
+            long utcMillis = TimeUnit.DAYS.toMillis(value);
 
-        // date is encoded as milliseconds at midnight in UTC
-        // convert to midnight if the session timezone
-        ISOChronology chronology = getChronology(session.getTimeZoneKey());
-        return utcMillis - chronology.getZone().getOffset(utcMillis);
+            // date is encoded as milliseconds at midnight in UTC
```

Comment:
This commit is from legacy implementation and it regards utcMillis variable in this context. It was quite clear to me what author meant when I was reading that first time, so I thought this comment is good enough in this legacy context.